enterprise/reporting/error_grouper.go

```diff
@@ -0,0 +1,47 @@
+package reporting
+
+import (
+	"github.com/samber/lo"
+
+	"github.com/rudderlabs/rudder-server/utils/types"
+)
+
+func (edr *ErrorDetailReporter) mergeMetricGroupsByErrorMessage(metricGroups map[types.ErrorDetailGroupKey][]*types.EDReportsDB) map[types.ErrorDetailGroupKey][]*types.EDReportsDB {
+	for groupKey, metrics := range metricGroups {
+		metricsGroupedByError := lo.GroupBy(metrics, func(metric *types.EDReportsDB) string {
+			return metric.ErrorMessage
+		})
+
+		mergedMetricsGroups := make(map[string]*types.EDReportsDB)
+		for errorMsg, metrics := range metricsGroupedByError {
+			mergedMetricsGroups[errorMsg] = metrics[0]
+			mergedMetricsGroups[errorMsg].Count = lo.SumBy(metrics, func(metric *types.EDReportsDB) int64 {
+				return metric.Count
+			})
+		}
+		metricGroups[groupKey] = lo.Values(mergedMetricsGroups)
+	}
+	return metricGroups
+}
```

Comment:
Since the grouping does not include event name, should we remove that label altogether from the `error_details_reports` table ? 

Let's say there are 3 track events, `product_viewed`, `product_added_to_cart`, `product_purchased` and all three have same error, we'll just count the events against one of these event names which might give the impression that all these events are let's say `product_viewed`